log/enabled.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package log // import "go.opentelemetry.io/otel/log"
+
+// EnabledParam represents payload for [Logger]'s Enabled method.
+type EnabledParam struct {
```

Comment:
non-blocking: personal preference for `EnabledOpts`.   It is just a naming pattern i've seen in other libraries (e.g. prometheus)

Reply:
I like the proposal. I had big trouble finding a good name.

Reply:
Renamed https://github.com/open-telemetry/opentelemetry-go/pull/5791/commits/450fc7a0c7eb084f8f7b39ee0a3b194f7ee254c7